tedge/src/cli/mqtt/subscribe.rs

```diff
@@ -29,37 +29,41 @@ impl Command for MqttSubscribeCommand {
 #[tokio::main]
 async fn subscribe(cmd: &MqttSubscribeCommand) -> Result<(), MqttError> {
     let config = cmd.mqtt_config.clone().clean_session();
-    let mqtt = Client::connect(cmd.client_id.as_str(), &config).await?;
     let filter = TopicFilter::new(cmd.topic.as_str())?.qos(cmd.qos);
 
-    let mut errors = mqtt.subscribe_errors();
-    let mut messages = mqtt.subscribe(filter).await?;
-
     loop {
-        select! {
-            error = errors.next().fuse() => {
-                if let Some(err) = error {
-                    if err.to_string().contains("MQTT connection error: I/O: Connection refused (os error 111)") {
-                        return Err(MqttError::ServerError(err.to_string()));
+        let mqtt = Client::connect(cmd.client_id.as_str(), &config).await?;
+        let mut errors = mqtt.subscribe_errors();
+        let mut messages = mqtt.subscribe(filter.clone()).await?;
+
+        loop {
+            select! {
+                maybe_error = errors.next().fuse() => {
+                    if let Some(error) = maybe_error {
+                        if error.to_string().contains("MQTT connection error: I/O: Connection refused (os error 111)") {
+                            return Err(MqttError::ServerError(error.to_string()));
+                        }
+                        async_println(&format!("ERROR: {:?}", error)).await?;
```

Comment:
Can't we silently re-connect instead of printing an error message?. I feel it's better to avoid printing this message.

Reply:
Since `tedge mqtt` is more like a dev tool, I'd prefer to see those error messages rather than silently ignoring it. But then, in that case I'd add one more word that we're `reconnecting` as well, in addition to printing that error message to let the user know that there was an error, but we're trying to recover out of it.